### ISSUE TITLE
strict-type-predicates: loosen when subject is in a type assertion

### ISSUE THREAD

**Issue by amacleay** [palantir/tslint#4467]
### Feature request

Related to https://github.com/palantir/tslint/issues/3972

**Is your feature request that we implement a new rule?**
Naw

**Is your feature request related to a problem? Please describe.**
`strict-type-predicates` fires when using the [textbook typeguard implementation](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards):

```ts
function isFish(pet: Fish | Bird): pet is Fish {
    return (<Fish>pet).swim !== undefined;  // strict-type-predicates   Expression is always true.
}
```

But it's only always true because I had to _assert_ it's true in order to get the compiler to loosen up and let me check the `swim` property!

**Describe the solution you'd like**
Do not fire when the type predicate expression includes a type assertion

**Describe alternatives you've considered**
A perfectly functional workaround is to apply `Partial` as well:

```ts
function isFish(pet: Fish | Bird): pet is Fish {
    return (<Partial<Fish>>pet).swim !== undefined;
}
```

This is just non-intuitive.

**Additional context**
Peep the discussion in #3972

**Comment by JoshuaKGoldberg** [comment]
The `Partial` case is actually the "correct" way of doing it, but strongly agreed it's not intuitive... perhaps the rule should suggest adding `Partial` in this case?

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
